### DISCUSSION TITLE
Break Change for p-autoComplete in 16.9.0

### DISCUSSION THREAD

**Discussion by vrjaroslav** [primefaces/community#555] [PrimeNG] [answered]
Hi, 

I use p-autoComplete. Like:

`<p-autoComplete (onSelect)="selectedProduct($event)"></p-autoComplete>`

and in TS: 
` 
selectedProduct(product: any) {
    console.log(product);  // here it immediately print the value, so the product  
}`

it was in version PrimeNg 15, but after I upgrade to version 16.9.0, then the method "onSelect" does not have directly selected value as parameter, but newly class: AutoCompleteOnSelectEvent, so I have change code to:

` 
selectedProduct(product: AutoCompleteOnSelectEvent) {
    console.log(product.value); 
}`

This is break change and I did not find anything about it anywhere. Also not here: [https://github.com/primefaces/primeng/blob/master/CHANGELOG.md]( https://github.com/primefaces/primeng/blob/master/CHANGELOG.md )

Do also have other components similar changes? 

Thank you for your help

**Comment by eXpertise7** [comment]
https://github.com/primefaces/primeng/issues/14018 I've already created a similar issue, and made one more discussion how migration docs/explanation are missing https://github.com/orgs/primefaces/discussions/162 here. I know people will encounter issues in their UI's after version updates, because not all stuff are documented.

**Comment by vrjaroslav** [comment]
thanks for reply. Yes, same issue. Actually I am worry that same break changes were made also on other components. It is impossible for me check documentation for all components or check code everywhere. Is it chance that somebody from PrimeNg team will try to update Migration Document please ?

**Reply by eXpertise7** [reply]
They do have: https://github.com/primefaces/primeng/releases , but not every breaking change is there.. Sometimes you'll have to figure out stuff on your own, like some of us do.

**Reply by vrjaroslav** [reply]
thanks @eXpertise7, if I will find any other similar issue, so I will report it here. It can also to somebody else

**Reply by eXpertise7** [reply]
https://github.com/primefaces/primeng/issues/14254 issue at 16.9.0 if you use `(onHide)` event from `p-sidebar`.

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz demo so we can identify the issue clearly?

**Reply by vrjaroslav** [reply]
Hi @cetincakiroglu , thank you for your answer. 

Sure, here is for PrimeNg 14: [https://stackblitz.com/edit/primeng-autocomplete-demo-flzhtx?file=src%2Fapp%2Fapp.component.ts](https://stackblitz.com/edit/primeng-autocomplete-demo-flzhtx?file=src%2Fapp%2Fapp.component.ts)

if you will input to autocompelte for example h-1 and then select it, you will see in console selected value

Here is link for PrimenNg 16: [https://stackblitz.com/edit/ysbwfs-99oprb?file=src%2Fapp%2Fdemo%2Fautocomplete-basic-demo.ts](https://stackblitz.com/edit/ysbwfs-99oprb?file=src%2Fapp%2Fdemo%2Fautocomplete-basic-demo.ts)

In this version if you input to autocomplete same value h-1 and selected it, so it will not print selected value, but value from: AutoCompleteOnSelectEvent.

So I need to change my code and newly use: event.value from AutoCompleteOnSelectEvent. And my question is if any other component has similar change

**Reply by vrjaroslav** [reply]
Hi @cetincakiroglu, 
the problem is between version 15 and 16, so you can ignore that Stackblitz for PrimeNg 14.  I created same for PrimeNg 15. Here is the link: [https://stackblitz.com/edit/rvfqb8-fcnxs8?file=src%2Fapp%2Fdemo%2Fautocomplete-basic-demo.ts](https://stackblitz.com/edit/rvfqb8-fcnxs8?file=src%2Fapp%2Fdemo%2Fautocomplete-basic-demo.ts)

**Comment by eXpertise7** [comment]
@vrjaroslav One example is `p-speedDial` component at primeng@15.2.0:

![image](https://github.com/primefaces/community/assets/17203747/7ea87673-0915-44cc-a4e3-2cee00c21b02)

```
    --item-diff-x: 2px;
    --item-diff-y: 2px;
```

are set like that, and `p-speedDial` appears normally.

But in primeng@16.9.0 (seems like the only stable version at Angular 16 - I've tested every version and every version had bugs, I'm updating Angular from 15 to 16 because of PrimeNG for 3 months already, process is: blocked and paused than in progress and again when PrimeNG releases new version):

![image](https://github.com/primefaces/community/assets/17203747/708b8b1f-1227-43d5-9ad7-70d3a0288535)

```
    --item-diff-x: 16px;
    --item-diff-y: 16px;
```

are set like that, and `p-speedDial` visually looks broken. PrimeNG Team has done change there, and I've to manually change this at my codebase/side to make stuff appear visually okay again (putting: ```
--item-diff-x: 2px !important;
    --item-diff-y: 2px !important;
``` with `:host ::ng-deep` selector at my component or main.scss file) - which is not a best practice.

There are at least 20+ changes I've seen that you'll have to adjust code at your CSS codebase at your project, stuff will break in TypeScript, in CSS, and in the HTML. **I'd advice you not to update to any primeng version less than 16.9.0**, only if you're lucky enough that everything works to you - that you use less components than I do, and that in some version everything works.

**Reply by vrjaroslav** [reply]
that's great, thank you @eXpertise7

**Reply by eXpertise7** [reply]
I'd suggest you to read PrimeNG docs at  their official website carefully, compare v14 and v16. And also, why updating 2 versions in the same time? Why not go from v14 to v15 (easy migration)? Than go from v15 to v16.9.0 (hard migration).

Read readme of release notes at their GitHub and read official docs carefully at Official website.

**Reply by vrjaroslav** [reply]
I did not migrate directly from 14 to 16. I migrated it first from 14 to 15 and then from 15 to 16. Yes, I read carefully the documentation

**Reply by eXpertise7** [reply]
Well, that was the reason I was surprised you mentioned PrimeNG 14. You should compare one before current major version (v15) with current stable one (v16). Kind regards.

**Comment by cetincakiroglu** [comment, marked as answer]
Mentioned issue for the autocomplete has been fixed and it doesn't exist in the v17
